.github/workflows/job_ci-checks.yml

```
@@ -0,0 +1,44 @@
+name: CI Checks
+
+on:
+  workflow_call:
+    inputs:
+      WORKING_DIRECTORY:
+        description: Path to working directory in repo
+        required: true
+        type: string
+
+jobs:
+  ci-checks:
+    name: Run CI checks
+    runs-on: ubuntu-24.04
+    env:
+      SAM_CLI_TELEMETRY: 0
+    defaults:
+      run:
+        shell: bash
+        working-directory: ${{ inputs.WORKING_DIRECTORY }}
+    steps:
+      - name: Check out repository code
+        uses: actions/checkout@11bd71901bbe5b1630ceea73d27597364c9af683 # v4.2.2
+        with:
+          submodules: true
+
+      - name: Setup nodeJS v20
```

nit: please remove v20. This is dynamic and may differ across workflows
```suggestion
      - name: Setup nodeJS
```